src/main/java/ch/uzh/ifi/access/student/evaluation/evaluator/CodeEvaluator.java

```diff
@@ -47,14 +47,26 @@ public SubmissionEvaluation evaluate(StudentSubmission submission, Exercise exer
 		validate(submission, exercise);
 		CodeSubmission codeSub = (CodeSubmission) submission;
 
-		SubmissionEvaluation.Points scoredPoints = parseScoreFromLog(codeSub.getConsole().getEvalLog());
-		List<String> hints = parseHintsFromLog(codeSub.getConsole().getEvalLog());
+		String log = codeSub.getConsole().getEvalLog();
+
+		SubmissionEvaluation.Points scoredPoints = parseScoreFromLog(log);
+		List<String> hints = parseHintsFromLog(log);
+
+		if (exercise.getMaxScore() != scoredPoints.getMax()) {
+			logger.info(String.format("Weird inconsistency: exercise says maxSore=%d, log parsing says maxScore=%d",
+					exercise.getMaxScore(), scoredPoints.getMax()));
+		}
+
+		int maxScore = Math.max(exercise.getMaxScore(), scoredPoints.getMax());
+		if (scoredPoints.getCorrect() == maxScore) {
+			hints.clear();
+		}
```

Comment:
I think it would be best to move the "max point check" before we go looking for hints. If we already know that all tests passed we can simply skip parsing strings .
To check if all tests passed you can check on `scoredPoints` if `correct == max`.

Reply:
So instead of clearing when the scores are equal, you would only call `parseHintsFromLog` in cases, where the scores are not equal?

Reply:
Yes exactly 👍